src/_pytest/compat.py

```diff
@@ -41,6 +42,19 @@ def _format_args(func):
 REGEX_TYPE = type(re.compile(""))
 
 
+if sys.version_info < (3, 6):
+
+    def fspath(p):
+        """os.fspath replacement, useful to point out when we should replace it by the
+        real function once we drop py35.
+        """
+        return str(p)
```

Comment:
this is not ~quite right -- but maybe good enough for our uses? here's the implementation from python3.6:

```python
def _fspath(path):
    """Return the path representation of a path-like object.

    If str or bytes is passed in, it is returned unchanged. Otherwise the
    os.PathLike interface is used to get the path representation. If the
    path representation is not str or bytes, TypeError is raised. If the
    provided path is not str, bytes, or os.PathLike, TypeError is raised.
    """
    if isinstance(path, (str, bytes)):
        return path

    # Work from the object's type to match method resolution of other magic
    # methods.
    path_type = type(path)
    try:
        path_repr = path_type.__fspath__(path)
    except AttributeError:
        if hasattr(path_type, '__fspath__'):
            raise
        else:
            raise TypeError("expected str, bytes or os.PathLike object, "
                            "not " + path_type.__name__)
    if isinstance(path_repr, (str, bytes)):
        return path_repr
    else:
        raise TypeError("expected {}.__fspath__() to return str or bytes, "
                        "not {}".format(path_type.__name__,
                                        type(path_repr).__name__))
```

Reply:
> but maybe good enough for our uses?

I think so. We will be using esporadically to convert `Path` objects only, and `__fspath__` of `Path` objects implement a call to `str()`:

```python
    def __fspath__(self):
        return str(self)
```

And it is also cached:

```python
    def __str__(self):
        """Return the string representation of the path, suitable for
        passing to system calls."""
        try:
            return self._str
        except AttributeError:
            self._str = self._format_parsed_parts(self._drv, self._root,
                                                  self._parts) or '.'
            return self._str
```

So I think it is fine as is. Also we will be dropping py35 soon I think, so we can just call `os.fspath` directly. 👍